app/src/org/commcare/android/database/connect/models/ConnectJobRecord.java

```diff
@@ -403,10 +405,20 @@ public Hashtable<String, Integer> getDeliveryCountsPerPaymentUnit(boolean todayO
         for(int i = 0; i < deliveries.size(); i++) {
             ConnectJobDeliveryRecord delivery = deliveries.get(i);
             int oldCount = 0;
-            if(paymentCounts.containsKey(delivery.getSlug())) {
-                oldCount = paymentCounts.get(delivery.getSlug());
+            boolean isSameDay=false;
+            if(todayOnly){
+                Date today = new Date();
+                isSameDay=sameDay(today,delivery.getDate());
+                if(paymentCounts.containsKey(delivery.getSlug()) && isSameDay) {
```

Comment:
Would be better not to duplicate this piece of code. Instead how about defining a boolean before the if-structure like:
boolean shouldInclude = true;

Then in the todayOnly case, you can update the value like:
shouldInclude = sameDay;

The else case goes away. Then after exiting the if-structure, you can do the following:
if(shouldInclude && paymentCounts.containsKey(delivery.getSlug()) ) {
   oldCount = paymentCounts.get(delivery.getSlug());
}